src/components/CharacterAssignment.jsx

```diff
@@ -0,0 +1,229 @@
+    }
+  })()}, [currentScript]);
+
+
+
+  // Reset the assigned characters to the actors so we don't have old character Ids interfering
+  const resetActors = () => {
+    const newColumns = {...columns}
+    for (let key in newColumns) {
+      newColumns[key].characterIds = [];
+      newColumns[key].lineCount = 0;
+    }
+    setColumns({...newColumns})
+  }
+
+
+  // Set all characters in their assigned column based on the fetched data
+  const setAssignedCharacters = (fetchedCharacters) => {
+    const newColumns = {...columns}
+
+    for (let key in fetchedCharacters){
+      let assignedActor = fetchedCharacters[key].actorId
+      if (assignedActor) {
+        newColumns[assignedActor].characterIds.push(fetchedCharacters[key].id)
+        newColumns[assignedActor].lineCount += fetchedCharacters[key].lineCount
+      } else {
+        newColumns.unassignedCharacters.characterIds.push(fetchedCharacters[key].id)
+        newColumns.unassignedCharacters.lineCount += fetchedCharacters[key].lineCount
+      }
+    }
+    setColumns({...newColumns})
+  }
+
+  // Create and update the columns for each of the actors and an Unassigned Characters column
+  const createOrUpdateActorColumns = (actors) => {
+    // Script is already set, only add or remove actors
+    if (currentScript && actors.length > 1  ) {
+      const newColumns = {...columns}
+      const newOrder = [...columnOrder]
+      // Adding an actor, there are more actors than columns (plus the unassigned column)
+      if (actors.length + 1 > newOrder.length){
+        actors.forEach((actor) => {
+          // Adding a new actor
+          if (!newColumns[actor.id]){
+            newColumns[actor.id] = {
+              id: actor.id.toString(),
+              title: actor.name,
+              characterIds: [],
+              lineCount: 0
+            }
+          newOrder.push(actor.id)
+          }
+        })
+      }
+      // Removing an actor, there are more columns than actors
+      else {
+        for (let key in newColumns){
+          // The current column doesn't match any actor or the unassigned column
+          if (!actors.some((actor) => actor.id == key) && key != 'unassignedCharacters'){
+          // Reassign characterIds to the unassigned column
+            const reassignedIds = newColumns[key].characterIds;
+            const reassignedCounts = newColumns[key].lineCount;
+            newColumns.unassignedCharacters.characterIds = [...newColumns.unassignedCharacters.characterIds, ...reassignedIds];
+            newColumns.unassignedCharacters.lineCount += reassignedCounts;
+            const index = newOrder.findIndex(id => id === key);
+            newOrder.splice(index, 1);
+          }
+        };
+      }
+      //Update the new column and order values after the actor changes
+      setColumns(newColumns)
+      setColumnOrder(newOrder)
+    }
+
+    // Initial setup of actor columns, no script is selected
+    else {
+    const actorColumns = {
```

Comment:
nit: run the code formatter to fix this indentation.  
Also, we generally don't want an `else` separated from the if block preceding it since it can cause devs to overlook their connection if they're not reading carefully enough.

Reply:
Ironically, in fixing this, I almost thought the else was part of a separate function.